teps/0139-trusted-artifacts.md

```diff
@@ -0,0 +1,477 @@
+---
+status: proposed
+title: Trusted Artifacts
+creation-date: '2023-07-27'
+last-updated: '2023-07-27'
+authors:
+- '@afrittoli'
+collaborators:
+- '@pritidesai'
+---
+
+# TEP-0139: Trusted Artifacts
+---
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+  - [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes and Caveats](#notes-and-caveats)
+- [Design Details](#design-details)
+- [Design Evaluation](#design-evaluation)
+  - [Reusability](#reusability)
+  - [Simplicity](#simplicity)
+  - [Flexibility](#flexibility)
+  - [User Experience](#user-experience)
+  - [Performance](#performance)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Implementation Plan](#implementation-plan)
+  - [Test Plan](#test-plan)
+  - [Infrastructure Needed](#infrastructure-needed)
+  - [Upgrade and Migration Strategy](#upgrade-and-migration-strategy)
+  - [Implementation Pull Requests](#implementation-pull-requests)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+The Tekton Data Interface working group has been working on for about 10 months now, identified a number of different problems to solve and proposed a number of different solutions.
+The number of different issues discussed and their sometimes conflicting requirements means that only a small fraction of the proposed solutions has actually been implemented in Tekton.
+This proposal is an attempt to take one of the problems identified, describe it in a way that is as much as possible self-contained, and provide a simple solution to it.
+The solution proposed does not need to address all the requirements and constrains of adjacent problems, but it should at least not make it harder for them to be addressed in future.
```

Comment:
Are there any specific adjacent problems you are referring to or just generally any adjacent problems that reviewers might have in mind?

Reply:
Good question!

Some examples of adjacent problems that we touched on in the working group are:

- pipeline level input artifacts (or materials)
- pipeline level output artifacts (or products)
- provide better data for provenance generation
- support non-PVC storage for artifacts
- support user-defined artifact storage strategies / steps
- optimize performance (data I/O, duration) and resources allocated for various use cases, like very short pipelines, very long pipelines, pipelines with high parallelism (e.g. matrix)
- provide a data I/O optimised solution for access to source repos